compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1109,353 +934,159 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
-                            INVALID_DOC_ATTRIBUTES,
-                            hir_id,
-                            meta.span,
-                            errors::DocAutoCfgHideShowExpectsList { attr_name },
-                        );
-                    }
-                }
-            }
-        }
-    }
-
     /// Runs various checks on `#[doc]` attributes.
     ///
     /// `specified_inline` should be initialized to `None` and kept for the scope
     /// of one item. Read the documentation of [`check_doc_inline`] for more information.
     ///
     /// [`check_doc_inline`]: Self::check_doc_inline
-    fn check_doc_attrs(
-        &self,
-        attr: &Attribute,
-        attr_span: Span,
-        style: AttrStyle,
-        hir_id: HirId,
-        target: Target,
-        specified_inline: &mut Option<(bool, Span)>,
-        aliases: &mut FxHashMap<String, Span>,
-    ) {
-        if let Some(list) = attr.meta_item_list() {
-            for meta in &list {
-                if let Some(i_meta) = meta.meta_item() {
-                    match i_meta.name() {
-                        Some(sym::alias) => {
-                            if self.check_attr_not_crate_level(meta, hir_id, "alias") {
-                                self.check_doc_alias(meta, hir_id, target, aliases);
-                            }
-                        }
-
-                        Some(sym::keyword) => {
-                            if self.check_attr_not_crate_level(meta, hir_id, "keyword") {
-                                self.check_doc_keyword_and_attribute(
-                                    meta,
-                                    hir_id,
-                                    DocFakeItemKind::Keyword,
-                                );
-                            }
-                        }
-
-                        Some(sym::attribute) => {
-                            if self.check_attr_not_crate_level(meta, hir_id, "attribute") {
-                                self.check_doc_keyword_and_attribute(
-                                    meta,
-                                    hir_id,
-                                    DocFakeItemKind::Attribute,
-                                );
-                            }
-                        }
-
-                        Some(sym::fake_variadic) => {
-                            if self.check_attr_not_crate_level(meta, hir_id, "fake_variadic") {
-                                self.check_doc_fake_variadic(meta, hir_id);
-                            }
-                        }
-
-                        Some(sym::search_unbox) => {
-                            if self.check_attr_not_crate_level(meta, hir_id, "fake_variadic") {
-                                self.check_doc_search_unbox(meta, hir_id);
-                            }
-                        }
+    fn check_doc_attrs(&self, attr: &DocAttribute, hir_id: HirId, target: Target) {
```

Comment:
Ooh I love how much nicer this file got, ty :3

Reply:
Less code always makes me happy too. :)